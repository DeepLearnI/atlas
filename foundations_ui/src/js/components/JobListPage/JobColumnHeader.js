import React, { Component } from 'react';
import PropTypes from 'prop-types';


class JobColumnHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: this.props.title,
      isStatus: this.props.isStatus,
    };
  }

  render() {
    const { title, isStatus } = this.state;
    let headerClassName = 'header-4 blue-border-bottom';
    let arrowClassName = 'arrow-down float-right';
    if (isStatus === 1) {
      headerClassName = 'header-4 blue-border-bottom status-header';
      arrowClassName = 'arrow-down margin-auto';
    }

    return (
      <div className="job-column-header">
        <h4 className={headerClassName}>{title}</h4>
        <div className={arrowClassName} />
      </div>
    );
  }
}

JobColumnHeader.propTypes = {
  title: PropTypes.string,
  isStatus: PropTypes.number,
};

JobColumnHeader.defaultProps = {
  title: '',
  isStatus: 0,
};

export default JobColumnHeader;
