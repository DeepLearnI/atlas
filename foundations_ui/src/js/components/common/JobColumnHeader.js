import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactTooltip from 'react-tooltip';
import JobListActions from '../../actions/JobListActions';

class JobColumnHeader extends Component {
  constructor(props) {
    super(props);
    this.onClickSort = this.onClickSort.bind(this);
    this.state = {
      title: this.props.title,
      isStatus: this.props.isStatus,
      offsetDivClass: this.props.className,
      containerDivClass: this.props.containerClass,
      mainHeader: this.props.mainHeader,
      isSortedColumn: this.props.isSortedColumn,
      isAscending: this.props.isAscending,
      sortTable: this.props.sortTable,
      selectAllJobs: this.props.selectAllJobs,
      allJobsSelected: this.props.allJobsSelected,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState(
      {
        title: nextProps.title,
        isSortedColumn: nextProps.isSortedColumn,
        isAscending: nextProps.isAscending,
        allJobsSelected: nextProps.allJobsSelected,
      },
    );
  }

  onClickSort() {
    const { sortTable, title, mainHeader } = this.state;

    sortTable(title, mainHeader);
  }

  render() {
    const {
      title, isStatus, offsetDivClass, containerDivClass, isSortedColumn,
      isAscending, selectAllJobs, allJobsSelected, mainHeader,
    } = this.state;
    const headerClassName = JobListActions.getJobColumnHeaderH4Class(isStatus);
    let divClassName = JobListActions.getJobColumnHeaderDivClass(containerDivClass, isStatus);

    let headerName = title;

    if (title === 'Tags') {
      divClassName = 'job-column-header tag-cell';
    }

    if (title === 'SelectAllCheckboxes') {
      headerName = <input type="checkbox" checked={allJobsSelected} onClick={() => { selectAllJobs(); }} />;
    }

    let arrowUp = null;
    let arrowDown = null;
    if (title !== '' && title.toLowerCase() !== 'job id' && title.toLowerCase() !== 'tags'
      && title !== 'SelectAllCheckboxes') {
      arrowUp = (
        <i
          onKeyPress={this.onClickSort}
          tabIndex={0}
          role="button"
          onClick={this.onClickSort}
          className={isSortedColumn && (isAscending === null || isAscending)
            ? 'i--icon-arrow-up' : 'i--icon-arrow-up-unfilled'}
        />
      );
      arrowDown = (
        <i
          onKeyPress={this.onClickSort}
          tabIndex={0}
          role="button"
          onClick={this.onClickSort}
          className={isSortedColumn && (isAscending === null || !isAscending)
            ? 'i--icon-arrow-down' : 'i--icon-arrow-down-unfilled'}
        />
      );
    }

    let dataClass = '';

    if (mainHeader === 'Metrics') {
      dataClass = 'metric-header';
    } else if (mainHeader === 'Parameters') {
      dataClass = 'param-header';
    }

    return (
      <div
        className={divClassName}
        ref={c => { this.headerContainer = c; }}
        data-class={dataClass}
      >
        <div className={offsetDivClass}>
          <h4
            className={`${headerClassName}`}
            data-tip={typeof (headerName) === 'string' && headerName.length > 15 ? headerName : ''}
          >
            {headerName}
            <ReactTooltip place="top" type="dark" effect="solid" />
          </h4>
          {arrowUp}
          {arrowDown}
        </div>
      </div>
    );
  }
}

JobColumnHeader.propTypes = {
  title: PropTypes.string,
  isStatus: PropTypes.bool,
  className: PropTypes.string,
  containerClass: PropTypes.string,
  toggleFilter: PropTypes.func,
  colType: PropTypes.string,
  isMetric: PropTypes.bool,
  isSortedColumn: PropTypes.bool,
  isAscending: PropTypes.bool,
  sortTable: PropTypes.func,
  selectAllJobs: PropTypes.func,
  allJobsSelected: PropTypes.bool,
  mainHeader: PropTypes.string,
};

JobColumnHeader.defaultProps = {
  title: '',
  isStatus: false,
  className: '',
  containerClass: 'job-column-header',
  toggleFilter: () => {},
  colType: 'string',
  isMetric: false,
  isSortedColumn: false,
  isAscending: false,
  sortTable: () => {},
  selectAllJobs: () => {},
  allJobsSelected: false,
  mainHeader: '',
};

export default JobColumnHeader;
