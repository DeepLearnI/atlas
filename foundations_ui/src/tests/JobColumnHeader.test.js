import React from 'react';
import ReactDOM from 'react-dom';
import JobColumnHeader from '../js/components/common/JobColumnHeader';
import { shallow, mount } from 'enzyme';
import configureTests from './setupTests';

configureTests();

it('Shallow Renders Job Column Header', () => {
  const wrapper = shallow(<JobColumnHeader/>);
  expect(wrapper).toMatchSnapshot();
});