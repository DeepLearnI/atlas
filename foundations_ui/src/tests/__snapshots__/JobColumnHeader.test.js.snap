// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Job Column Header 1`] = `
<div
  className="job-column-header"
>
  <div
    className=""
  >
    <h4
      className="header-4 blue-border-bottom"
    />
    <div
      className="arrow-down float-right"
    />
  </div>
</div>
`;
